_posts/2019-11-08-method-references-the-unpolished-gem-of-java-8.markdown

```diff
@@ -280,7 +280,7 @@ final class EvenBetterDogValidator {
 
     final boolean isAGoodBoy(final Dog dog) = dogValidator::isAGoodBoy;
 
-    final boolean isABadBoy(final Dog dog) = this:not:isAGoodBoy;
+    final boolean isABadBoy(final Dog dog) = Predicates:not:isAGoodBoy;
```

Comment:
Since there's no `not` method in `this`, I believe you wanted to use the `not` from `Predicates` that you mentioned earlier

Reply:
Nope, this is OK: `this::isAGoodBoy` is a method reference of type `Predicate<Dog>` so we should be able to negate it by putting `not` in the middle there (:

Reply:
What about: `not:this:isAGoodBoy`? or just: `not:isAGoodBoy`? Ideally, it could be good, but I see this syntactically complicated... not impossible, but at least I feel confused about who owns the `not` method (only looking at the syntax).

Reply:
> `not:this:isAGoodBoy`
That works but conflicts with my earlier proposal.
> not:isAGoodBoy
Can't do, reference needs a source object / class :(
> I see this syntactically complicated
Agreed it might not be easy to implement!
This design is definitely flawed and many things could be improved about it.

My goal with this is to start a discussion more than show a working example (;